packages/hooking/manager.ts

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright 2022 Code Intelligence GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+	Hook,
+	HookType,
+	HookFn,
+	BeforeHookFn,
+	ReplaceHookFn,
+	AfterHookFn,
+} from "./hook";
+
+export type MatchingHooksResult = {
```

Comment:
Perhaps it would make the code easier to understand if `MatchingHooksResult` is a real class with methods to check if specific hook types are set.